CMakeLists.txt

```diff
@@ -11,21 +19,20 @@ option(KokkosFFT_INTERNAL_Kokkos "Build internal Kokkos instead of relying on ex
 option(KokkosFFT_ENABLE_BENCHMARK "Build benchmarks for KokkosFFT" OFF)
 
 # Version information
-set(KokkosFFT_VERSION_MAJOR 0)
-set(KokkosFFT_VERSION_MINOR 0)
-set(KokkosFFT_VERSION_PATCH 00)
-set(KokkosFFT_VERSION "${KokkosFFT_VERSION_MAJOR}.${KokkosFFT_VERSION_MINOR}.${KokkosFFT_VERSION_PATCH}")
-
-#Set variables for config file
-math(EXPR KOKKOSFFT_VERSION "${KokkosFFT_VERSION_MAJOR} * 10000 + ${KokkosFFT_VERSION_MINOR} * 100 + ${KokkosFFT_VERSION_PATCH}")
+math(EXPR KOKKOSFFT_VERSION "${PROJECT_VERSION_MAJOR} * 10000 + ${PROJECT_VERSION_MINOR} * 100 + ${PROJECT_VERSION_PATCH} * 10 + ${PROJECT_VERSION_TWEAK}")
 math(EXPR KOKKOSFFT_VERSION_MAJOR "${KOKKOSFFT_VERSION} / 10000")
 math(EXPR KOKKOSFFT_VERSION_MINOR "${KOKKOSFFT_VERSION} / 100 % 100")
 math(EXPR KOKKOSFFT_VERSION_PATCH "${KOKKOSFFT_VERSION} % 100")
 
+set(KOKKOS_REQUIRED_VERSION "4.2.00")
+
 if (NOT KokkosFFT_INTERNAL_Kokkos)
     # First check, Kokkos is added as subdirectory or not
     if(NOT TARGET Kokkos::kokkos)
         find_package(Kokkos REQUIRED)
+        # Kokkos version check
+        include(${CMAKE_CURRENT_SOURCE_DIR}/cmake/Kokkos_Version_Check.cmake)
```

Comment:
Since you update `CMAKE_CURRENT_SOURCE_DIR`, you don't need to specify the absolute path anymore and can simply do:

```cmake
include(Kokkos_Version_Check)
```

Reply:
Do they work in both KokkosFFT as library and KokkosFFT as subdirectory cases?

Reply:
This file is not included in the installation of Kokkos FFT, or is it?

Moreover, if you need a specific version of Kokkos, you should pass it as `REQUIRED` in `find_package` (if this is possible).

Reply:
For build based on pre-installed Kokkos, this is included. 

> Moreover, if you need a specific version of Kokkos, you should pass it as REQUIRED in find_package (if this is possible, see @cedricchevalier19 comment).

As commented, I would like custom error messages.